File: backend/app/schemas/linkedin.py
from pydantic import BaseModel
from typing import List, Optional


class LinkedInImportRequest(BaseModel):
    url: Optional[str] = None
    text: Optional[str] = None


class SectionSuggestion(BaseModel):
    text: str
    boost: str  # e.g. "+12"


class LinkedInSection(BaseModel):
    id: str
    label: str
    score: int
    current: str
    weaknesses: List[str]
    suggestions: List[SectionSuggestion]


class LinkedInAnalysis(BaseModel):
    base_score: int
    sections: List[LinkedInSection]
    missing_skills: List[str]
    quick_wins: List[str]


class GenerateSuggestionsRequest(BaseModel):
    section_id: str
    current_content: str
    target_role: Optional[str] = None

File: frontend/lib/api/linkedin.ts
import { request } from "./client";

export interface SectionSuggestion {
  text: string;
  boost: string;
}

export interface LinkedInSection {
  id: string;
  label: string;
  score: number;
  current: string;
  weaknesses: string[];
  suggestions: SectionSuggestion[];
}

export interface LinkedInAnalysis {
  base_score: number;
  sections: LinkedInSection[];
  missing_skills: string[];
  quick_wins: string[];
}

export const linkedinApi = {
  import: (data: { url?: string; text?: string }) =>
    request<Record<string, unknown>>("/linkedin/import", {
      method: "POST",
      body: JSON.stringify(data),
    }),

  analyze: (profileData: Record<string, unknown>) =>
    request<LinkedInAnalysis>("/linkedin/analyze", {
      method: "POST",
      body: JSON.stringify(profileData),
    }),

  generateSuggestions: (sectionId: string, currentContent: string, targetRole?: string) =>
    request<SectionSuggestion[]>("/linkedin/generate-suggestions", {
      method: "POST",
      body: JSON.stringify({
        section_id: sectionId,
        current_content: currentContent,
        target_role: targetRole,
      }),
    }),
};
